applications/StructuralMechanicsApplication/custom_elements/solid_elements/small_displacement_mixed_strain_displacement_element.h

```diff
@@ -0,0 +1,766 @@
+      */
+    void SetValuesOnIntegrationPoints(
+        const Variable<double>& rVariable,
+        const std::vector<double>& rValues,
+        const ProcessInfo& rCurrentProcessInfo
+        ) override;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    const Parameters GetSpecifications() const override;
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        std::stringstream buffer;
+        buffer << "Small Displacement Mixed Strain Element #" << Id() << "\nConstitutive law: " << mConstitutiveLawVector[0]->Info();
+        return buffer.str();
+    }
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "Small Displacement Mixed Strain Element #" << Id() << "\nConstitutive law: " << mConstitutiveLawVector[0]->Info();
+    }
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override
+    {
+        pGetGeometry()->PrintData(rOStream);
+    }
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+protected:
+
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    IntegrationMethod mThisIntegrationMethod;                     /// Integration method
+    std::vector<ConstitutiveLaw::Pointer> mConstitutiveLawVector; /// The vector containing the constitutive laws
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    /**
+     * @brief Sets the used integration method
+     * @param ThisIntegrationMethod Integration method used
+     */
+    void SetIntegrationMethod(const IntegrationMethod& ThisIntegrationMethod)
+    {
+        mThisIntegrationMethod = ThisIntegrationMethod;
+    }
+
+    /**
+     * @brief Sets the used constitutive laws
+     * @param ThisConstitutiveLawVector Constitutive laws used
+     */
+    void SetConstitutiveLawVector(const std::vector<ConstitutiveLaw::Pointer>& ThisConstitutiveLawVector)
+    {
+        mConstitutiveLawVector = ThisConstitutiveLawVector;
+    }
+
+    /**
+     * @brief It initializes the material
+     */
+    virtual void InitializeMaterial();
+
+    /**
+     * @brief This method returns if the element provides the strain
+     */
+    virtual bool UseElementProvidedStrain() const;
+
+    /**
+     * @brief This method returns the stabilization factor tau according to the FE size
+     */
+    double GetStabilizationFactor() const
+    {
+        const auto &r_props = GetProperties();
+        const auto &r_geom = GetGeometry();
+        const SizeType dim = r_geom.WorkingSpaceDimension();
+        const double tau = r_props.Has(STABILIZATION_FACTOR) ? r_props[STABILIZATION_FACTOR] : default_stabilization_factor; // tau is c / L0
+        if (r_props.Has(CHARACTERISTIC_LENGTH_MULTIPLIER)) {
```

Comment:
Is `CHARACTERISTIC_LENGTH_MULTIPLIER` the element size appearing in the stabilisation constants? (I guess so). If so, I'd use the `ElementSizeUtilities` rather than asking for it or computing it in here.

Reply:
Any comment on this?